tests/gpu/kmeans/test_kmeans_h2o4gpu.cu

```diff
@@ -0,0 +1,41 @@
+#include "gtest/gtest.h"
+
+#include "../src/gpu/kmeans/kmeans_h2o4gpu.h"
+#include "../src/gpu/kmeans/kmeans_labels.h"
+#include <thrust/host_vector.h>
+
+TEST(KMeans, CountsPerCentroids) {
+  // GIVEN
+  int k = 2;
+  int d = 2;
+  int n = 4;
+
+  // Setup data
+  thrust::host_vector<float> h_data(n*d);
+  h_data[0] = 0.0f; h_data[1] = 0.0f; // [0,0]
+  h_data[2] = 0.0f; h_data[3] = 0.0f; // [0,0]
+  h_data[4] = 1.0f; h_data[5] = 1.0f; // [1,1]
+  h_data[6] = 1.0f; h_data[7] = 1.0f; // [1,1]
+  thrust::device_vector<float> *d_data = new thrust::device_vector<float>();
```

Comment:
It is best practice to never manually allocate memory (new/malloc) in c++11. This device vector can just be a local variable. This means it is on the stack and gets freed at the end of scope. There are two memory leaks here as new is called without calling delete.